File: app-cpp/src/file-retrieval-engine.cpp
#include <iostream>

#include "IndexStore.hpp"
#include "RetrievalEngine.hpp"
#include "ApplicationInterface.hpp"

int main(int argc, char** argv)
{
    std::shared_ptr<IndexStore> store = std::make_shared<IndexStore>();
    std::shared_ptr<RetrievalEngine> engine = std::make_shared<RetrievalEngine>(store);
    std::shared_ptr<ApplicationInterface> interface = std::make_shared<ApplicationInterface>(engine);

    interface->readCommands();

    return 0;
}

File: app-java/src/main/java/csc435/app/FileRetrievalEngine.java
package csc435.app;

public class FileRetrievalEngine 
{
    public static void main( String[] args )
    {
        IndexStore store = new IndexStore();
        RetrievalEngine engine = new RetrievalEngine(store);
        ApplicationInterface appInterface = new ApplicationInterface(engine);

        appInterface.readCommands();
    }
}
